src/pages/patientView/trialMatch/TrialMatchTableUtils.ts

```diff
@@ -0,0 +1,129 @@
+import {
+    IArmMatch, IClinicalGroupMatch, IDetailedTrialMatch, IGenomicGroupMatch, IGenomicMatch, ITrial,
+    ITrialMatch, IArm, IDrug
+} from "../../../shared/model/MatchMiner";
+import * as _ from 'lodash';
+
+export function groupTrialMatchesById(trials: ITrial[], trialMatches:ITrialMatch[]): IDetailedTrialMatch[] {
+    const matchesGroupedById = _.groupBy(trialMatches, (trial: ITrialMatch) => trial.id);
+    let matchedTrials: IDetailedTrialMatch[] = [];
+    _.forEach(matchesGroupedById, (trialGroup, trialId) => {
+        const originalMatchedTrial: ITrial = _.find( trials, { 'id': trialId } )!;
+        let matchedTrial:IDetailedTrialMatch = {
+            id: originalMatchedTrial.id,
+            nctId: originalMatchedTrial.nctId,
+            protocolNo: originalMatchedTrial.protocolNo,
+            phase: originalMatchedTrial.phase,
+            shortTitle: originalMatchedTrial.shortTitle,
+            status: originalMatchedTrial.status,
+            matches: [],
+            priority: 0 // highest priority
+        };
+        groupTrialMatchesByArmDescription(trialGroup, matchedTrial, originalMatchedTrial);
+        matchedTrials.push(matchedTrial);
+    });
+    return _.sortBy(matchedTrials, (trial: IDetailedTrialMatch) => trial.priority);
+}
+
+export function groupTrialMatchesByArmDescription(trialGroup: ITrialMatch[], matchedTrial: IDetailedTrialMatch, originalMatchedTrial: ITrial) {
```

Comment:
functions should always return something, even if it's just mutating data passed in.

Reply:
Added `return` statement for all functions.